test/parse-test.js

```diff
@@ -58,6 +58,18 @@ var expectations = {
 
 beforeEach(function (){
   Mustache.clearCache();
+  Mustache.templateCache = {
+    _cache: {},
+    set: function set (key, value) {
+      this._cache[key] = value;
+    },
+    get: function get (key) {
+      return this._cache[key];
+    },
+    clear: function clear () {
+      this._cache = {};
+    }
+  };
```

Comment:
Hmz, I'm a bit concerned this will prevent us from testing the default built-in cache found in the source (`mustache.mjs`). Cause if we do some adjustments to that default cache sometime in the future, those adjustments will be overwritten by this custom template cache, right?

Guessing this trick is to reset the template cache back to (what's currently) the default cache, especially important after the custom template cache test has run, right?

If so, maybe it would be possible to:

1. get a hold of the *actual default* template cache in a `before()`
2. store a reference to that original template cache object in a top-level variable
3. stick that reference back as part of this `beforeEach()`, something like `Mustache.templateCache = originalTemplateCache`

..or have I totally misunderstood the purpose of this?

Reply:
Done, I had a quick look at the eslint docs to see if there was a rule to enforce this. But the closest is `no-undef` which also stops `undefined` from being used to initialise objects etc. So I have raised a new rule proposal in the ESLint repo: https://github.com/eslint/eslint/issues/12754

Reply:
You understood correctly, good catch. I have updated this, had to provide a getter for the Writer `templateCache` as well to facilitate this.